### PR TITLE
feat(benchpress): Add extension for ff metrics reporting

### PR DESCRIPTION
**Pull request by hankduan**

### CODE REVIEW

**Comment by hankduan** [comment]
The failures are because it's creating dart files from these js files. I'm not sure why it's doing that.

File: modules/benchpress/test/util/firefox_extension/helper.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+var q = require('q');
+var FirefoxProfile = require('firefox-profile');
+var pathUtil = require('path');
+var fs = require('fs');
+
+exports.getFirefoxProfile = function(extensionPath) {
```
Comment (tbosch) [inline modules/benchpress/test/util/firefox_extension/helper.js]: This file should go under `src/firefox_extension` as people need to use this when running benchpress with firefox.

File: package.json [hunk 2]
```diff
@@ -59,6 +59,7 @@
     "dgeni": "^0.4.1",
     "dgeni-packages": "^0.10.11",
     "event-stream": "^3.1.5",
+    "firefox-profile": "0.3.4",
```
Comment (tbosch) [inline package.json]: needs to be a real dependency, not just a devDependency, so people that use benchpress can use the `helper` as well.

Comment (tbosch) [inline package.json]: Please also add an entry to the `package.json` of benchpress itself: See https://github.com/angular/angular/blob/master/modules/benchpress/package.json

File: modules/benchpress/test/util/firefox_extension/spec.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    // Run some commands
+    element(by.model('yourName')).sendKeys('Hank');
+    expect(element(by.binding('yourName')).getText()).toEqual('Hello Hank!');
+
+    browser.executeScript('window.forceGC()');
+
+    var script = 
+        'window.stopAndRecord("' + browser.params.profileSavePath + '")';
+    browser.executeScript(script).then(function() {
```
Comment (tbosch) [inline modules/benchpress/test/util/firefox_extension/spec.js]: Could you make at least one assertions for the result of the profile?

**Comment by tbosch** [comment]
Let's leave the xpi in the sources for now, so we can get this change in as soon as possible. We can change the build later on.

**Comment by tbosch** [comment]
Oh, wait, just found [jpm](https://developer.mozilla.org/en-US/Add-ons/SDK/Tools/jpm), a node.js based replacement for `cfx`. Could you use this instead of the bash script, and just create the extension on the fly on every testrun, so we don't need to build/store it at all?

**Comment by tbosch** [comment]
@hankduan Could you rename your files into `.es6`? Then Dart won't pick them up. Also, please remove the `util` folder in between and put the files under `src//firefox_extension` and `test//firefox_extension`.

Otherwise LGTM.

File: modules/benchpress/test/util/firefox_extension/helper.js (hunk 1 above)
Comment (tbosch) [inline modules/benchpress/test/util/firefox_extension/helper.js]: Use jpm directly in here so we don't need to build/store the extension at all.

Comment (tbosch) [inline modules/benchpress/test/util/firefox_extension/helper.js]: Something like `require('jpm/lib/xpi')(packageJson, {xpiPath: someTempPath});`

**Comment by tbosch** [comment]
Oh, and we probably should run this test on every build as well :-)
Add a line [here](https://github.com/angular/angular/blob/master/scripts/ci/test_e2e_js.sh#L32) that starts a new protractor instance with your configuration and your test.

File: package.json (hunk 2 above)
Comment (hankduan) [inline package.json]: done.

File: modules/benchpress/test/util/firefox_extension/spec.js (hunk 3 above)
Comment (hankduan) [inline modules/benchpress/test/util/firefox_extension/spec.js]: added a simplistic assertion that just checks for the existence of the file and that it contains the gc entry. Obviously needs more work, but it'll be easier once the converter is done.

**Comment by hankduan** [comment]
I fixed all the mentioned issues, and renamed all the js files to es6. 

Can you talk a little about (or point me to the code with) the mechanism that angular uses to transform es6 to js? i.e. does it transform all the es6 files, or certain es6 files?
Part of this will help me understand https://github.com/angular/angular/blob/master/scripts/ci/test_e2e_js.sh#L32. i.e. it is referring to protractor-js.conf.js, but what's actually creating those files as they're not in the repo.

**Comment by tbosch** [comment]
Hank,
we are transpiling our sources into the `dist` folder, from where our `protractor-js.conf` picks it up (see https://github.com/angular/angular/blob/master/protractor-shared.js#L114).

**Comment by tbosch** [comment]
Hank, I merged your PR as I want to typscript'ify benchpress tomorrow. Please integrate the e2e test in another PR. Btw, do we still need the `.gitingore` fire?

**Comment by hankduan** [comment]
We can remove 
addon-sdk*

but we would still need
*.xpi

**Pull request closed without merging**